lib/sles4sap/sap_deployment_automation_framework/configure_sap_systems_tfvars.pm

```diff
@@ -0,0 +1,442 @@
+settings. Data structure converted into tfvars format is written into target file.
+
+=over
+
+=item * B<av_zones>: Define if availability zones should be used. Default: true
+
+=item * B<workload_vnet_code>: Workload zone VNET code.
+
+=back
+
+=cut
+
+sub create_sap_systems_tfvars {
+    my (%args) = @_;
+    # use AV zones by default
+    $args{av_zones} //= 'true';
+    my $sap_sid = get_required_var('SAP_SID');
+
+    croak 'Missing mandatory argument "$args{workload_vnet_code}"' unless $args{workload_vnet_code};
+
+    # Keep deployment scenario mandatory without defaults to keep setting consistent across all test modules
+    my @components = split(',', get_required_var('SDAF_DEPLOYMENT_SCENARIO'));
+    validate_components(components => \@components);
+
+    my $hana_ha = grep(/db_ha/, @components) ? 'true' : 'false';
+    my $ensa2_ha = grep(/nw_ensa/, @components) ? 'true' : 'false';
+    my $app_deployment = grep(/ensa|pas|aas|scs/, @components) ? 'true' : 'false';
+    my $app_server_count = grep(/pas/, @components) + grep(/aas/, @components);
+    my $scs_server_count = $app_deployment eq 'true' ? '1' : '0';
+
+    my %tfvars_data;
+    $tfvars_data{file_header} = "### File was generated by OpenQA automation according to template:\n### https://github.com/Azure/SAP-automation-samples/blob/main/Terraform/WORKSPACES/SYSTEM/LAB-SECE-SAP04-L00/LAB-SECE-SAP04-L00.tfvars\n";
+    $tfvars_data{env_definitions} = define_sap_systems_environment(
+        environment => get_required_var('SDAF_ENV_CODE'),
+        location => get_required_var('PUBLIC_CLOUD_REGION'),
+        resource_group => generate_resource_group_name(deployment_type => 'sap_system'),
+        bom_name => get_required_var('SDAF_BOM_NAME')
+    );
+    $tfvars_data{sap_systems_networking} = define_networking(workload_vnet_code => $args{workload_vnet_code});
+    $tfvars_data{cluster_settings} = define_cluster_settings();
+    $tfvars_data{database_tier} = define_database_tier(high_availability => $hana_ha, av_zones => $args{av_zones});
+    $tfvars_data{application_tier} = define_application_tier(sap_sid => $sap_sid,
+        enable_app_tier_deployment => $app_deployment);
+    $tfvars_data{sap_central_services} = define_sap_central_services(
+        scs_high_availability => $ensa2_ha, scs_server_count => $scs_server_count, sap_sid => $sap_sid);
+    $tfvars_data{application_servers} = define_application_servers(
+        app_server_count => $app_server_count, sap_sid => $sap_sid);
+    $tfvars_data{miscellaneous_settings} = define_miscellaneous_settings();
+    $tfvars_data{nfs_support} = define_nfs_settings();
+    $tfvars_data{vm_images} = define_vm_images(os_image => get_required_var('PUBLIC_CLOUD_IMAGE_ID'));
+
+    my $tfvars_file = get_os_variable('sap_system_parameter_file');
+    write_tfvars_file(tfvars_data => \%tfvars_data, tfvars_file => $tfvars_file);
+    upload_logs($tfvars_file, log_name => 'sap_systems.tfvars.txt');
```

Comment:
is it `.txt` added to be able to open it directly within the browser?

Reply:
Yes. That was already in old code. It's just moved here from deleted lib.

Reply:
> is it `.txt` added to be able to open it directly within the browser?

I think yes.